# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GCRuntime.h
# Commit: 28829816a629
# Full Hash: 28829816a629b28125db3b4c65c5e22518e167e8
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-09 21:37:17
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 6: Stop using JSContext for GC helper thread tasks r=jandem
#   
#   This patch moves the debug-only GC state from JSContext to JSFreeOp. Support
#   functions for this move to FreeOp.h and most RAII classes move to GCInternals.h.
#   
# ==============================================================================

diff -r 410d1b57bb09 -r 28829816a629 js/src/gc/GCRuntime.h
--- a/js/src/gc/GCRuntime.h	Wed Mar 09 10:17:07 2022 +0000
+++ b/js/src/gc/GCRuntime.h	Wed Mar 09 10:17:08 2022 +0000
@@ -14,6 +14,7 @@
 
 #include "gc/ArenaList.h"
 #include "gc/AtomMarking.h"
+#include "gc/FreeOp.h"
 #include "gc/GCMarker.h"
 #include "gc/IteratorUtils.h"
 #include "gc/Nursery.h"
@@ -919,6 +920,8 @@
   // Embedders can use this zone however they wish.
   MainThreadData<JS::Zone*> systemZone;
 
+  MainThreadData<JSFreeOp> mainThreadFreeOp;
+
  private:
   // All zones in the runtime, except the atoms zone.
   MainThreadOrGCTaskData<ZoneVector> zones_;